### PR TITLE
DM-14760: Merge StorageFileInfo into StorageInfo

### PR DESCRIPTION
**Pull request by pschella**

### CODE REVIEW

File: python/lsst/daf/butler/core/storedFileInfo.py [hunk 1]
```diff
@@ -58,6 +58,10 @@ def __init__(self, formatter, path, storageClass):
         self._path = path
         assert isinstance(storageClass, StorageClass)
         self._storageClass = storageClass
+        assert checksum is None or isinstance(checksum, str)
```
Comment (TallJimbo) [inline python/lsst/daf/butler/core/storedFileInfo.py:61]: `str` or `bytes`?

File: python/lsst/daf/butler/core/storedFileInfo.py [hunk 2]
```diff
@@ -76,3 +80,19 @@ def storageClass(self):
...
+
+    @property
+    def size(self):
+        """Size of stored object in bytes (`int`).
+        """
+        return self._size
+
+    def __repr__(self):
+        return f"{type(self).__qualname__}(path=\"{self.path}\", formatter=\"{self.formatter}\"" \
+            f" size={self.size}, checksum=\"{self.checksum}\", storageClass=\"{self.storageClass.name}\")"
```
Comment (TallJimbo) [inline python/lsst/daf/butler/core/storedFileInfo.py]: This is pretty hard to read with all the escaping - seems like a solid reason to use single-quotes in some of these places.

**Review by TallJimbo — APPROVED** [review]
Looks good.  Only really minor comments.

File: python/lsst/daf/butler/core/storedFileInfo.py (hunk 1 above)
Comment (timj) [reply python/lsst/daf/butler/core/storedFileInfo.py:61]: Currently `computeChecksum` returns a hex digest as a str. We don't have byte checksums flying around.

**Pull request merged**